src/app/_mermaid_common/libs/directives/leaflet-map.dir.js

```diff
@@ -34,11 +34,35 @@ angular.module('mermaid.libs').directive('leafletMap', [
         scope.mapopts = scope.mapopts || {};
         scope.records = scope.records || [];
         scope.geoattr = scope.geoattr || 'location';
-        scope.maprecords = L.geoJson([], {
+
+        var mapRecordsProperty = {
           pointToLayer: function(feature, latlng) {
             return new L.circleMarker(latlng, style);
           }
-        });
+        };
+
+        if (scope.mapopts.showPopup) {
+          mapRecordsProperty.onEachFeature = function(feature, layer) {
+            var featureProperties = feature.properties;
+            layer.bindPopup(
+              '<a href="#/projects/' +
+                featureProperties.project_id +
```

Comment:
Here and below (`rec.name` etc.) can we code it so that this generic map directive doesn't use hardcoded properties, i.e. assume that it's being used for sites?

Also, we should protect against nonexistent properties and handle gracefully.